src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -123,7 +123,7 @@ public virtual TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext)
                 metadata,
                 name,
                 value: "true",
-                useViewData: !explicitValue,
+                useViewData: (metadata == null && !explicitValue),
```

Comment:
Also a different variable name? There's a parameter `isExplicitValue` that follows which always has a false value. Reusing it this name seems messy.